src/status.rs

```diff
@@ -247,44 +295,117 @@ pub fn show_build_status(
 
 		let now = Instant::now();
 
-		if estimation_impossible || simulated_time <= now {
-			println!(
-				"Building for {}, estimating remaining time...\x1b[K\x1b[m",
-				MinSec::since(start_time)
-			);
+		let current_duration = start_time.elapsed();
+		let remaining_duration = if estimation_impossible {
+			None
+		} else if simulated_time < now {
+			// Should have been done already, estimate it will be done immediately
+			Some(Duration::from_millis(0))
 		} else {
-			let remaining_duration = simulated_time - now;
-			let eta = TimeDuration::from_std(remaining_duration)
-				.map(|duration| chrono::Local::now() + duration);
-			match eta {
-				Ok(eta) => {
-					let is_soon = eta.date() == chrono::Local::now().date()
-						&& remaining_duration < Duration::from_secs(8 * 3600);
-					let timeformat = if is_soon {
-						"%H:%M:%S"
+			Some(simulated_time - now)
+		};
+
+		let (progress, percentagetext, remainingtext, etatext) = match remaining_duration {
+			None => (
+				0.,
+				"??".to_owned(),
+				"estimating time".to_owned(),
+				"estimating time".to_owned(),
+			),
+			Some(remaining_duration) => {
+				let progress = as_millis(current_duration) as f64
+					/ as_millis(current_duration + remaining_duration) as f64;
+				(
+					progress,
+					format!("{:02}%", (progress * 100.).ceil() as u8),
```

Comment:
`.ceil()`? In other places (like the progress bar itself) we use `floor`. Now you display `100%` for the full last percent, instead of only when it's completely done.